### ISSUE TITLE
HitBTC3 Python

### ISSUE THREAD

**Issue by DavidSteinhart** [ccxt/ccxt#10959]
I tried using the CCXT HitBTC3 library with python and found the following issues:

1. Cannot call private functions because of error in line 1475 
headers['Authorization'] = 'HS256 ' + encoded -> should probably be 
headers['Authorization'] = b'HS256 ' + encoded

2. Precision for price and mount is reversed in line 403 - 405
'precision': {'price': lot,'amount': step,},

3. parse_order does not parse symbol and sets id to client_id when using fetch_open_orders

```
import ccxt

apiKey = ""
secret = ""
hitbtc3 = getattr(ccxt, 'hitbtc3')({'apiKey': apiKey, 'secret': secret})
hitbtc3.load_markets()

# 1.
hitbtc3.fetch_balance()
#   File "C:\Program Files\Python39\lib\site-packages\ccxt\hitbtc3.py", line 562, in fetch_balance
#     response = self.privateGetSpotBalance(params)
#   File "C:\Program Files\Python39\lib\site-packages\ccxt\base\exchange.py", line 497, in inner
#     return entry(_self, **inner_kwargs)
#   File "C:\Program Files\Python39\lib\site-packages\ccxt\base\exchange.py", line 546, in request
#     return self.fetch2(path, api, method, params, headers, body, config, context)
#   File "C:\Program Files\Python39\lib\site-packages\ccxt\base\exchange.py", line 541, in fetch2
#     request = self.sign(path, api, method, params, headers, body)
#   File "C:\Program Files\Python39\lib\site-packages\ccxt\hitbtc3.py", line 1475, in sign
#     headers['Authorization'] = 'HS256 ' + encoded
# TypeError: can only concatenate str (not "bytes") to str

# 2.
print(hitbtc3.markets['BTC/USDT']['precision'])
# {'price': 1e-05, 'amount': 0.01} -> should be the other way round

# 3.
print(hitbtc3.fetch_open_orders())
# [{'info': {'id': '123', 'client_order_id': 'abc', 'symbol': 'KEYUSDT', 'side': 'buy', 'status': 'new', 'type': 'limit', 'time_in_force': 'GTC', 'quantity': '430', 'quantity_cumulative': '0', 'price': '0.01141965', 'post_only': False, 'created_at': '2021-12-20T13:20:47.226Z', 'updated_at': '2021-12-20T13:20:47.226Z'}, 'id': 'abc', 'clientOrderId': 'abc', 'timestamp': 1640006447226, 'datetime': '2021-12-20T13:20:47.226Z', 'lastTradeTimestamp': None, 'symbol': None, 'price': Decimal('0.01141965'), 'amount': Decimal('430.0'), 'type': 'limit', 'side': 'buy', 'timeInForce': 'GTC', 'postOnly': False, 'filled': 0.0, 'remaining': 430.0, 'cost': 0.0, 'status': 'open', 'average': None, 'trades': [], 'fee': None, 'fees': []}]

# -> symbol is None
# -> id = clinetOrderId
```

**Comment by DavidSteinhart** [comment]
I just looked at the fix and I think there is still an error. "lot" is used with price for precision but with amount for limits. So assigning
```
const lotString = this.safeString (market, 'quantity_increment');
const stepString = this.safeString (market, 'tick_size');
```
was correct but then in precision lot and step should be switched.

```
'precision': {
   'price': lot, # should be step
   'amount': step # should be lot
}
'limits': {
   'amount': {
      'min': lot,
      'max': None,
   },
   'price': {
      'min': step,
      'max': None,
   },
}
```

**Comment by DavidSteinhart** [comment]
Really really sorry to bother again, but now it was switched twice, so 

```
const lotString = this.safeString (market, 'tick_size');
const stepString = this.safeString (market, 'quantity_increment');
```
should be switched to

```
const lotString = this.safeString (market, 'quantity_increment');
const stepString = this.safeString (market, 'tick_size');
```
and then it should work.

**Comment by kroitor** [comment]
@DavidSteinhart pardon for the inconvenience, pushed another fix, it will arrive in ~20 minutes in 1.64.48. Thanks for your attention!
